brickstrap.sh

```diff
@@ -404,7 +440,10 @@ case "${cmd}" in
     copy-root)           copy-root;;
     configure-packages)  configure-packages;;
     run-hook)            run-hook ${BOARDDIR}/hooks/${run_hook_arg};;
-    run-hooks)           run-hooks;;
+    run-hooks)
+        run-hooks
+        dump-info-hook
```

Comment:
> Since this is run at the same time as other hooks, why not just use a hook for this?

Because it is fundamentally not meant to be the same kind of hook: i.e. the script is not supposed to do arbitrary things but just log relevant info. So it makes more sense (to me) to  simply defer running of the hook to the right point in time, which brickstrap knows, and therefore guarantee a well-defined execution environment.

Also, the way the hooks are currently set up simply naming this specially will not work:
 * The hooks are not sorted by name before being executed in-order. The order will therefore be (on Linux) whatever `*` expansion returns them in. So not stable/predictable. Therefore it would be fragile.
 * Additionally fixing this is not easy to do cleanly because by design the hooks expect to be sourced (so simply executing them through run-parts won't work).

Beyond that hooks run before image creation, and if you wish to log things about the image (e.g. SHA256 sums) you need to run... after.

Deferring the hook until when it is actually meant to be invoked solves all this by design.